control-plane/csi-driver/src/bin/node/volume_serializer.rs

```diff
@@ -0,0 +1,85 @@
+use once_cell::sync::OnceCell;
+use snafu::Snafu;
+use std::{collections::HashSet, sync::Mutex};
+use tracing::{error, trace};
+
+static VOLUME_SERIALIZER: OnceCell<Mutex<HashSet<String>>> = OnceCell::new();
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub(crate)), context(suffix(false)), module(nvme_error))]
+/// Possible volume serializer error kind sources.
+pub(crate) enum VolumeSerializerError {
+    #[snafu(display("VOLUME_SERIALIZER once cell is not accessible"))]
+    OnceCellInaccessible,
+    #[snafu(display("Nvme connect operation is in progess for volume :{}", id))]
+    OperationInProgess { id: String },
+}
+
+impl From<VolumeSerializerError> for tonic::Status {
+    fn from(error: VolumeSerializerError) -> Self {
+        match error {
+            VolumeSerializerError::OnceCellInaccessible => {
+                tonic::Status::unavailable(error.to_string())
+            }
+            VolumeSerializerError::OperationInProgess { .. } => {
+                tonic::Status::already_exists(error.to_string())
+            }
+        }
+    }
+}
+
+/// Initialise the VOLUME_SERIALIZER once cell.
+pub(crate) fn init_volume_serializer(data: HashSet<String>) {
```

Comment:
data here has no meaning, I suggest you remove this init function completely, and simply initialize it on the gtter function by calling directly get_or_init, example:
```rust
/// Returns the Volume Serializer hash set.
pub(crate) volume_serializer() -> &'static Mutex<HashSet<String>> {
    VOLUME_SERIALIZER.get_or_init(|| Mutex::new(HashSet::new()))
}
```
Also this way there's no error case :)

Reply:
Great suggestion. Thanks. Done.